omod/src/main/java/org/openmrs/module/appui/rest/SessionController.java

```diff
@@ -47,6 +47,16 @@ public Object get(HttpServletRequest request, HttpServletResponse response) {
         return ret;
     }
 
+    @RequestMapping(method = RequestMethod.POST)
+    @ResponseBody
+    public Object setLocation(@RequestParam("location") Location location,
```

Comment:
I think it would also be best to base this on our rest web services module's framework